src/audio/copier/copier.c

```diff
@@ -1491,6 +1491,13 @@ static int copier_params(struct comp_dev *dev, struct sof_ipc_stream_params *par
 			if (dev->ipc_config.type == SOF_COMP_HOST && !cd->ipc_gtw) {
 				component_set_nearest_period_frames(dev, params->rate);
 				if (params->direction == SOF_IPC_STREAM_CAPTURE) {
+					enum sof_ipc_frame __sparse_cache dummy, out_ipc_fmt;
+
+					audio_stream_fmt_conversion(cd->out_fmt->depth,
+								    cd->out_fmt->valid_bit_depth,
+								    &dummy, &out_ipc_fmt,
+								    cd->out_fmt->s_type);
+					params->frame_fmt = out_ipc_fmt;
```

Comment:
This change is good but why we have "if (params->direction == SOF_IPC_STREAM_CAPTURE) {" here. It was introduced by host copier optimization.  The params only affect host or dai, why not move it to the entry of copier_params like
````c
static int copier_params(struct comp_dev *dev, struct sof_ipc_stream_params *params)
{
	struct copier_data *cd = comp_get_drvdata(dev);
	struct comp_buffer *sink, *source;
	struct comp_buffer __sparse_cache *sink_c, *source_c;
	struct list_item *sink_list;
	int ret = 0;
	int i;

	comp_dbg(dev, "copier_params()");

	memset(params, 0, sizeof(*params));
	params->direction = cd->direction;
	params->channels = cd->config.base.audio_fmt.channels_count;
	params->rate = cd->config.base.audio_fmt.sampling_frequency;

+ if (params->direction == SOF_IPC_STREAM_PLAYBACK) {
+	params->sample_container_bytes = cd->config.base.audio_fmt.depth / 8;
+	params->sample_valid_bytes = cd->config.base.audio_fmt.valid_bit_depth / 8;
+      	params->buffer.size = cd->config.base.ibs;
+ } else {
+      params->buffer.size = cd->config.base.obs;
+     params->sample_container_bytes = cd->out_fmt->depth / 8;
+     params->sample_valid_bytes =
+		cd->out_fmt->valid_bit_depth / 8;
+}
	params->stream_tag = cd->config.gtw_cfg.node_id.f.v_index + 1;
	params->frame_fmt = dev->ipc_config.frame_fmt;
	params->buffer_fmt = cd->config.base.audio_fmt.interleaving_style;


````